packages/eas-cli/src/metadata/config.ts

```diff
@@ -0,0 +1,42 @@
+import Ajv from 'ajv';
+import assert from 'assert';
+
+import { AppleConfigReader } from './apple/config/reader';
+import { AppleConfigWriter } from './apple/config/writer';
+import { AppleMetadata } from './apple/types';
+
+export interface MetadataConfig {
+  /** The store configuration version */
+  configVersion: number;
+  /** All App Store related configuration */
+  apple?: AppleMetadata;
+}
+
+/**
+ * Run the JSON Schema validation to normalize defaults and flag early config errors.
+ * This includes validating the known store limitations for every configurable property.
+ */
+export function validateConfig(config: unknown): {
+  valid: boolean;
+  errors: Ajv.ErrorObject[];
+} {
+  const validator = new Ajv({ allErrors: true, useDefaults: true })
+    .addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'))
+    .compile(require('../../schema/metadata-0.json'));
+
+  const valid = validator(config) as boolean;
+
+  return { valid, errors: validator.errors || [] };
+}
+
+/** Create a versioned deserializer to fetch App Store data from the store configuration. */
+export function createAppleReader(config: MetadataConfig): AppleConfigReader {
+  assert(config.configVersion === 0, 'Unsupported store configuration version');
+  assert(config.apple, 'No apple configuration found');
```

Comment:
assert should be used to mitigate programmer's error, if you want to handle the error that might happen because the user e.g. select  incorrect option then it should throw an exception

Reply:
I'm not sure if I understand this. You are right that these would be user-configuration errors, but I'm not sure why those should be thrown errors.

From [the docs](https://nodejs.org/api/assert.html):
> The `node:assert` module provides a set of assertion functions for verifying invariants.

With React, for example, the user could be returning the wrong underlying component, causing invariant errors, right?

Reply:
assert should not be used for error handling, few examples

bad use cases:
 - user-specified an invalid value
 - network request failed
 - API returned an error

 good use cases:
 - if you know that you validated sth in the past but type does not reflect that (e.g. after refactor assert might fail, which shows a bug in a code)
 - if API response returns sth in a wrong format and I'm not talking (already a bit questionable)
 - vary rare errors e.g. out of disk out of memory (also questionable)
 
 https://en.wikipedia.org/wiki/Assertion_(software_development)

> an assertion is a [predicate](https://en.wikipedia.org/wiki/Predicate_(mathematical_logic)) (a [Boolean-valued function](https://en.wikipedia.org/wiki/Boolean-valued_function) over the [state space](https://en.wikipedia.org/wiki/State_space), usually expressed as a [logical proposition](https://en.wikipedia.org/wiki/Logical_proposition) using the [variables](https://en.wikipedia.org/wiki/Variable_(programming)) of a program) connected to a point in the program, that always should evaluate to true at that point in code execution. Assertions can help a programmer read the code, help a [compiler](https://en.wikipedia.org/wiki/Compiler) compile it, or help the program detect its own defects.